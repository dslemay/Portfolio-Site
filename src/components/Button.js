import React from 'react';
import PropTypes from 'prop-types';
import Link from 'gatsby-link';
import styled from 'react-emotion';
import { css } from 'emotion';

const ButtonStyles = props => css`
  display: block;
  background-color: ${props.theme.complementaryLight};
  color: ${props.theme.baseColor};
  padding: 0.2rem 1.3rem;
  border-radius: 20px;
  text-decoration: none;
  cursor: pointer;
`;

const AnchorTag = styled.a`
  ${ButtonStyles};
`;

const LinkTag = styled(Link)`
  ${ButtonStyles};
`;

const BasicButton = styled.button`
  ${ButtonStyles};
`;

const Button = props => {
  if (props.href) {
    return (
      <AnchorTag href={props.href} target="_blank">
        {props.children}
      </AnchorTag>
    );
  }

  if (props.to) {
    return <LinkTag to={props.to}>{props.children}</LinkTag>;
  }

  return <BasicButton type={props.type}>{props.children}</BasicButton>;
};

Button.defaultProps = {
  type: 'button'
};

Button.propTypes = {
  href: PropTypes.string,
  to: PropTypes.string,
  type: PropTypes.oneOf(['button', 'submit'])
};

export default Button;
