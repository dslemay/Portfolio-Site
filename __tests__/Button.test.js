import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button';

describe('Button', () => {
  it('should render an anchor element if passed an href property', () => {
    const wrapper = mount(<Button href="http://dslemay.com" />);
    expect(wrapper.find('a').length).toBe(1);
  });

  it('renders properly when passed an href property', () => {
    const tree = renderer.create(<Button href="http://dslemay.com" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an anchor element if passed a to property', () => {
    const wrapper = mount(<Button to="http://dslemay.com" />);
    expect(wrapper.find('MockedLink').length).toBe(1);
  });

  it('renders properly when passed a to property', () => {
    const tree = renderer.create(<Button to="http://dslemay.com" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a basic button if not passed any props', () => {
    const wrapper = mount(<Button />);
    expect(wrapper.find('button').length).toBe(1);
  });
});
