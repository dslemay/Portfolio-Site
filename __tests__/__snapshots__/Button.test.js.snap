// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly when passed a to property 1`] = `
<MockedLink
  className="css-1n2cjf4 css-1egae7x1"
  to="http://dslemay.com"
/>
`;

exports[`Button renders properly when passed an href property 1`] = `
<a
  className="css-1n2cjf4 css-1egae7x0"
  href="http://dslemay.com"
  target="_blank"
/>
`;
